File: output/gpt4/java/Solution02_gpt.java
public class Solution02_gpt {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        
        int reversed = 0;
        int original = x;
        
        while (original > 0) {
            reversed = reversed * 10 + original % 10;
            original /= 10;
        }
        
        return x == reversed;
    }
}

File: output/gpt4/js/Solution02_gpt.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if (x < 0) {
        return false;
    }
    
    let reversed = 0;
    let original = x;
    
    while (original > 0) {
        reversed = reversed * 10 + original % 10;
        original = Math.floor(original / 10);
    }
    
    return x === reversed;
};